src/common/filter/Expressions.h

```diff
@@ -313,6 +313,10 @@ class Expression {
         return std::abs(left - right) < EPSILON;
     }
 
+    static bool contains(const std::string& left, const std::string& right) {
```

Comment:
is there any case that right is empty string?

Reply:
It's possible.